nft_ingester/src/program_transformers/bubblegum/creator_verification.rs

```diff
@@ -67,9 +67,12 @@ where
                 let tree_id = cl.id.to_bytes();
                 let nonce = cl.index as i64;
 
+                // Start a db transaction.
+                let multi_txn = txn.begin().await?;
```

Comment:
this is too late to start the TXN. The TXN should include `asset_should_be_updated` and `asset_should_be_updated` should be using SELECT ... FOR UPDATE instead of the SELECT only. This would allow you to lock the rows you are selecting and fail transactions  that cause a raise condition.

Reply:
Understood, I see how the txn could solve issue with asset_should_be_updated